helm/argocd-application/templates/application.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: {{ include "argocd_application.fullname" . }}
+  labels:
+    {{- include "argocd_application.labels" . | nindent 4 }}
+spec:
+  project: {{ .Values.project }}
+  {{ with .Values.source }}
+  source:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{ with .Values.destination }}
+  destination:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{ with .Values.syncPolicy }}
+  syncPolicy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{ with .Values.info }}
+  info:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
```

Comment:
Why don't we use whole blob for configuration template ?

Reply:
It's easier to have it in this format when we want to reuse the same blob for Kubernetes manifest object. It also handles all of the possible cases for this object so unless the object definition changes, it should have the same result.